src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
Agreed the later scenario is the most relevant: have the ability to execute scoped `IConfigureOptions` without having to pay for them on every request. 

What I am most interested in revisiting is the reasons we originally decided we needed to create a complete new interface in 1.1 instead of adding behavior to `IOptions`. Talking through what you would need to do to observe any difference in behavior with @ajcvickers this afternoon it seemed like it is very contrived.